app/assets/javascripts/host_edit.js

```diff
@@ -363,11 +366,14 @@ function architecture_selected(element){
 }
 
 function os_selected(element){
-  var attrs = attribute_hash(['operatingsystem_id', 'organization_id', 'location_id']);
   var url = $(element).attr('data-url');
+  var type = $(element).attr('data-type');
+  var obj = (type == "hosts" ? "host" : "hostgroup");
+  var attrs = {};
+  attrs[obj] = attribute_hash(['operatingsystem_id', 'organization_id', 'location_id']);
```

Comment:
And monkey patch this in each plugin? I don't like that. Global array seems a better idea than this :-)